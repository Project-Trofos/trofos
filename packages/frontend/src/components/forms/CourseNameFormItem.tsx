import React from 'react';
import { Form, Input } from 'antd';

// Renders form elements to input course name
export default function CourseNameFormItem(): JSX.Element {
  return (
    <Form.Item
      label="Course Name"
      name="courseName"
      rules={[
        { required: true, message: "Please input your course's name!" },
        { pattern: /^[a-zA-Z0-9-]*$/, message: 'The course name must be alphanumeric.' },
        { max: 64, message: 'The key must be at most 64 characters long.' },
      ]}
    >
      <Input />
    </Form.Item>
  );
}
