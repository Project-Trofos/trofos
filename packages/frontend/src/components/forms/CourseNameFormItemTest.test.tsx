import React from 'react';
import { render, screen } from '@testing-library/react';
import FormWrapper from './__testutils__/FormRenderHelper.test';
import CourseNameFormItem from './CourseNameFormItem';

describe('CourseNameFormItem test', () => {
  function renderForm() {
    return render(
      <FormWrapper>
        <CourseNameFormItem />
      </FormWrapper>,
    );
  }

  describe('when rendering', () => {
    it('should have correct fields', () => {
      const { baseElement } = renderForm();
      expect(baseElement).toMatchSnapshot();

      expect(screen.getByLabelText(/course name/i)).toBeInTheDocument();
    });
  });
});
