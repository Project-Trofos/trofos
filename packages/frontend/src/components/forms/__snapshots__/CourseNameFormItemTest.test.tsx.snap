// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CourseNameFormItem test > when rendering > should have correct fields 1`] = `
<body>
  <div>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label"
          >
            <label
              class="ant-form-item-required"
              for="courseName"
              title="Course Name"
            >
              Course Name
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input ant-input-outlined"
                  id="courseName"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
        type="submit"
      >
        <span>
          submit
        </span>
      </button>
    </form>
  </div>
</body>
`;
